addons/account_debit_note/models/account_journal.py

```diff
@@ -0,0 +1,24 @@
+from odoo import api, fields, models
+
+
+class AccountJournal(models.Model):
+    _inherit = "account.journal"
+
+    debit_sequence = fields.Boolean(
+        string="Dedicated Debit Note Sequence",
+        help="Check this box if you don't want to share the same sequence for invoices "
+        "and debit notes made from this journal",
+        default=False,
+    )
+
+    @api.onchange("type")
+    def _onchange_type(self):
+        super()._onchange_type()
+        self.debit_sequence = self.type in ("sale", "purchase")
+
+    @api.model
+    def _fill_missing_values(self, vals, protected_codes=False):
```

Comment:
@jco-odoo This feels like outdated. Can we change it to stored, compute? 
https://github.com/odoo/odoo/blob/852c1235a60cf08f8d8f81ec5256f288a92ad63c/addons/account/models/account_journal.py#L555-L558

https://github.com/odoo/odoo/blob/852c1235a60cf08f8d8f81ec5256f288a92ad63c/addons/account/models/account_journal.py#L803-L806

Reply:
@hamo-odoo It gave me the same idea.  I think so.  api.depends('type') then